src/python/pants/core/goals/lint.py

```diff
@@ -46,29 +45,17 @@ def prep_output(s: bytes) -> str:
         )
 
 
-class LinterFieldSet(FieldSetWithOrigin, metaclass=ABCMeta):
-    """The fields necessary for a particular linter to work with a target."""
-
-    sources: Sources
-
-
-_FS = TypeVar("_FS", bound="LinterFieldSet")
-
-
 @union
-class LinterFieldSets(Collection[_FS]):
-    """A collection of `FieldSet`s for a particular linter, e.g. a collection of
-    `Flake8FieldSet`s."""
-
-    field_set_type: ClassVar[Type[_FS]]
+class LintRequest(StyleRequest):
+    """A union for StyleRequests that should be lintable."""
```

Comment:
It would be good to explain that this still corresponds 1-1 with a particular linter.